skfem/mesh/mesh.py

```diff
@@ -84,6 +84,68 @@ def subdomains(self):
     def boundaries(self):
         return self._boundaries
 
+    def encode_point_data(self):
```

Comment:
Here is the relevant part.

Reply:
Will the point-data work for &lsquo;boundaries&rsquo; which aren't on the boundary? Currently the facet-based approach does support this (as does Gmsh in exporting (*n* –1)-dimensional cells representing facets).

Reply:
Old issue on internal boundaries: #336

Reply:
A preliminary investigation (based on the MWE from #683)
```python
m = MeshLine([0, 1, 2])
m.define_boundary("mid1", lambda x: x[0] == 1.0, boundaries_only=False)
print("Before:", m.boundaries)
print("After:", from_meshio(to_meshio(m)).boundaries)

```
suggests no: the internal boundary `"mid1"`of `m` is emptied during the round-trip `to_meshio`, `from_meshio`:
> Before: {'mid1': array([1])}
> After: {'mid1': array([], dtype=int64)}

Reply:
Of course that is an improvement on the master branch, in which `to_meshio` doesn't attempt to serialize the boundaries at all.

> Before: {'mid1': array([1])}
> /home/gdmcbain/src/scikit-fem/skfem/io/meshio.py:134: UserWarning: Unable to load tagged boundaries/subdomains.
  warnings.warn("Unable to load tagged boundaries/subdomains.")
> 'gmsh:physical'
> After: None

Reply:
I think the culprint is `_decode_point_data`:

https://github.com/kinnala/scikit-fem/blob/d0febca6118745fd41c4bb04679efec40fb3f167/skfem/mesh/mesh.py#L109-L110

Reply:
The approach in this PR cannot support "interfaces". I also think we should not call them "boundaries" anymore. However, Mesh.interfaces will require a different approach. I propose that we do that in a different issue/PR, I'm running out of time on this one.

Reply:
I mean, suppose you have a tetrahedral element with all vertices on the boundary. If `point_data` contains the value 1 on all of its vertices, then is the interior facet part of the named boundary or not? In order to serialize internal facets, we should write to `cell_data`, some matching identified to both elements sharing the facet.

Reply:
>  I propose that we do that in a different issue/PR, I'm running out of time on this one.

Sure, sorry.  Press ahead and read this later if at all.

> If point_data contains the value 1 on all of its vertices, then is the interior facet part of the named boundary or not?

It could be that this indicates a limitation of the idea of representing boundaries or interfaces by points when really they're facets of domain-cells.

> In order to serialize internal facets, we should write to cell_data a matching identifier for both elements sharing the interior facet.

Actually the approach taken in #681 seems to cope O. K. with "interfaces".  It doesn't distinguish between interfaces and boundaries; the `.boundaries` attribute of `skfem.Mesh` is just an array of indices of `.facets` which belong to the named subset, as previously.  Because it works directly with the facets of the internal representation, this just works.

> The approach in this PR cannot support "interfaces". I also think we should not call them "boundaries" anymore. 

Are they indeed distinct?  Up till now it seems to have been O. K. to just treat them as named subsets of facets.

>  In order to serialize internal facets, we should write to cell_data a matching identifier for both elements sharing the interior facet.

That's right, that is what was done automatically in #681, because such a facet showed up as `True` in two columns of the `isin` array:  

https://github.com/kinnala/scikit-fem/blob/cfb38f7212eb1cde93b7686480b30f5921adfc63/docs/examples/ex_261.py#L18-L23

But you're running out of time and I'm quibbling, push ahead and yes this can be sorted out later if required.  I can use #681 in-house indefinitely.